mdagent/tools/base_tools/analysis_tools/rmsd_tools.py

```diff
@@ -2,314 +2,192 @@
-        np.savetxt(
-            f"{self.path_registry.ckpt_records}/{csv_filename}",
-            rmsf_data,
-            delimiter=",",
-            header="Residue_ID,RMSF",
-            comments="",
-        )
-        message = f"Saved RMSF data to {csv_filename}.\n"
-        self.path_registry.map_path(
-            csv_filename, f"{self.path_registry.ckpt_records}/{csv_filename}", message
-        )
+    if traj_fileid not in all_fileids:
+        raise ValueError("Trajectory File ID not found in path registry")
 
-        # Plot RMSF
-        if plot:
-            plt.figure(figsize=(5, 3))
-            plt.plot(atoms.resnums, rmsf, label=str(selection))
-            plt.xlabel("Residue Number")
-            plt.ylabel("RMSF ($\AA$)")
-            plt.title("Root Mean Square Fluctuation")
-            plt.legend()
-            fig_name = self.path_registry.write_file_name(
-                type=FileType.FIGURE,
-                fig_analysis=self.filename,
-                file_format="png",
-            )
-            plt.savefig(f"{self.path_registry.ckpt_figures}/{fig_name}")
-            plot_message = f"Plotted RMSF. Saved to {fig_name}.\n"
-            message += plot_message
-            self.path_registry.map_path(
-                f"{self.filename}.png", f"{self.filename}.png", plot_message
-            )
-        return message
+    traj_path = path_registry.get_mapped_path(traj_fileid)
+    return md.load(traj_path, top=top_path)
 
 
-class RMSDInputSchema(BaseModel):
-    rmsd_type: str = Field(
-        description="""type of RMSD calculation
-        to perform. Choose from 'rmsd', 'pairwise_rmsd', 'rmsf'.
-        'rmsd': any 1-D root mean square deviation calculations.
-        'pairwise_rmsd': 2D root mean square deviation calculation.
-            pairwise RMSD matrix is computed. Either
-            trajectory against itself or a given reference.
-        'rmsf': root mean square fluctuation. it computes the average
-            fluctuation for each residue for the entire trajectory.
-        """
+def save_to_csv(path_registry, data, file_id, description=None):
+    file_path = f"{path_registry.ckpt_figures}/{file_id}.csv"
+    i = 0
+    while os.path.exists(file_path):
+        i += 1
+        file_path = f"{path_registry.ckpt_figures}/{file_id}_{i}.csv"
+    np.savetxt(file_path, data, delimiter=",")
+    path_registry.map_path(file_id, file_path, description=description)
+    return file_path
+
+
+def save_plot(path_registry, fig_analysis, description=None):
+    fig_name = path_registry.write_file_name(
+        type=FileType.FIGURE,
+        fig_analysis=fig_analysis,
+        file_format="png",
+    )
+    fig_id = path_registry.get_fileid(file_name=fig_name, type=FileType.FIGURE)
+    fig_path = f"{path_registry.ckpt_figures}/{fig_name}"
+    plt.savefig(fig_path)
+    path_registry.map_path(fig_id, fig_path, description=description)
+    return fig_id, fig_path
```

Comment:
These are functions we will use in multiple places, can you generalize them and move them to utils?

Reply:
done :)